src/cascadia/TerminalSettingsModel/BaseVisualStudioGenerator.h

```diff
@@ -22,20 +22,19 @@ namespace Microsoft::Terminal::Settings::Model
 {
     class BaseVisualStudioGenerator : public IDynamicProfileGenerator
     {
-        virtual bool IsInstanceValid(const VsSetupConfiguration::VsSetupInstance instance) const = 0;
-        virtual std::wstring GetProfileName(const VsSetupConfiguration::VsSetupInstance instance) const = 0;
-        virtual std::wstring GetProfileCommandLine(const VsSetupConfiguration::VsSetupInstance instance) const = 0;
-        virtual std::wstring GetProfileGuidSeed(const VsSetupConfiguration::VsSetupInstance instance) const = 0;
-        virtual std::wstring GetProfileIconPath() const = 0;
-
+    public:
         // Inherited via IDynamicProfileGenerator
-        virtual std::wstring_view GetNamespace() override = 0;
+        std::wstring_view GetNamespace() override = 0;
         std::vector<winrt::Microsoft::Terminal::Settings::Model::Profile> GenerateProfiles() override;
 
-    private:
-        inline static bool hasQueried = false;
-        inline static std::vector<VsSetupConfiguration::VsSetupInstance> instances;
+    protected:
+        virtual bool IsInstanceValid(const VsSetupConfiguration::VsSetupInstance& instance) const = 0;
+        virtual std::wstring GetProfileName(const VsSetupConfiguration::VsSetupInstance& instance) const = 0;
+        virtual std::wstring GetProfileCommandLine(const VsSetupConfiguration::VsSetupInstance& instance) const = 0;
+        virtual std::wstring GetProfileGuidSeed(const VsSetupConfiguration::VsSetupInstance& instance) const = 0;
+        virtual std::wstring GetProfileIconPath() const = 0;
 
+    private:
         winrt::Microsoft::Terminal::Settings::Model::Profile CreateProfile(const std::wstring_view instanceId);

```

Comment:
```suggestion
        winrt::Microsoft::Terminal::Settings::Model::Profile _CreateProfile(const std::wstring_view instanceId);
```

Reply:
All names beginning with an underscore are reserved for the language or the C/C++ library!

Reply:
You can just keep it that way @DHowett. I'm going to inline that function in my PR, because I'm changing the type to `implementation::Profile` anyways.